src/ReservoirPriceOracle.sol

```diff
@@ -164,9 +160,10 @@ contract ReservoirPriceOracle is IPriceOracle, IReservoirPriceOracle, Owned(msg.
 
     // IReservoirPriceOracle
 
+    // REVIEW: This let's anyone read the Reservoir oracle without paying right?
```

Comment:
that's right. We have two options at this point: 

1. allow everyone to use it first, then monetize if it even gains any traction
2. build a gatekeeper now, will cost more gas for all queries.

Recommendation is to go with 1 for now 

see discussion [here](https://discord.com/channels/1039538523646205952/1062163738431602758/1245702886843748435)

Reply:
Isn't there a 3rd option, ReservoirOracle is the only caller for now. If we want to migrate we wipe all routes and set the new oracle/chainlink as fallback?

Reply:
We can upgrade that's correct. Then if needed the access controls can be enforced on the new oracle. Also then `Shared::OracleCaller` can be set to this new oracle as well. 
```
                            pairs
                              ^
                              |  queries
vault --> old oracle --> new oracle 
```

But there will still be a loophole. Because new oracle would always allow calls from old oracle, and that old oracle's access is not controlled, anyone can still call old oracle. There's no way to know who's calling old oracle since from the new oracle's perspective the `msg.sender` will be old oracle.

For the loophole mentioned above, it'd just be for the `getQuotes` function, since that's the call getting forwarded. With the upgrade you suggested, not everyone would be able to call the `getTimeWeightedAverage`.

Reply:
Right but I guess we can just remove getTimeWeightedAverage in the interest of simplicty and focus 100% on getting euler working. Yes technically non-euler contracts can use getQuotes but i agree with you that locking down the oracle is not of critical importance.

So I'm now suggesting we remove `getTimeWeightedAverage` to reduce line count + api footprint

Reply:
alright makes sense. Then we should remove `getLatest` as well since that's the instantaneous version for oracle observations. I'll also get rid of `IReservoirPriceOracle` since without those 2 function, it won't exist anymore. 

Will implement

Reply:
done